lib/daru/index.rb

```diff
@@ -24,7 +24,7 @@ def self.new *args, &block
       source = args[0]
 
       idx =
-      if source and source[0].is_a?(Array)
+      if source.respond_to?(:first) && source.first.is_a?(Array)
```

Comment:
I still don't understand the reason behind replacing the previous line with this one. Also, don't squash commits until the PR is ready. Once the commit is gone, the comments made on it cannot be seen.